Update 2fa script a bit more + generate 4.1 changelog.

Carreau opened the pull request:
The number of commits /stats will be off by a few, but not sure it is really that important.

Carreau committed:
Fix 2fa for gh_stats tools.

takluyver commented on tools/gh_api.py:
```diff
@@ -17,7 +17,7 @@
     import requests_cache
 except ImportError:
-    print("no cache", file=sys.stderr)
+    print("cache, not available, install request_cache for caching.", file=sys.stderr)
```
requests_cache

takluyver replied:
also, no comma after 'cache'

Carreau replied:
fixed.

takluyver commented on tools/github_stats.py:
```diff
@@ -212,8 +212,7 @@ def report(issues, show_urls=False):
     if milestone:
-        print("The full list can be seen `on GitHub <https://github.com/%s/milestone/%s>`__"
-            % (project, milestone)
+        print("The full list can be seen `on GitHub <https://github.com/{project}/issues?q=milestone%3A{milestone}+>`__".format(project=project,milestone=milestone)
```
Getting ready for f strings? ;-)

Carreau replied:
:-P

takluyver commented:
The code changes look fine to me, though I kind of think that since we're doing stats, they should be created at release time so that they're accurate.

Carreau commented:
> The code changes look fine to me, though I kind of think that since we're doing stats, they should be created at release time so that they're accurate.

fixed, rebased.
